torch/_inductor/lowering.py

```diff
@@ -348,6 +352,27 @@ def transform_args(
             args[args_indices[0]] if args_indices else kwargs[kwargs_indices[0]]
         ).get_device()
 
+        def _is_copy_cpu_scalar(x: TensorBox) -> TensorBox:
+            size = x.get_size()
+            cur_device = x.get_device()
+            if (
+                cur_device is not None
+                and cur_device.type == "cpu"
+                and cur_device != device
+                and isinstance(x.data, ir.ReinterpretView)
+                and (len(size) == 0 or (len(size) == 1 and size[0] == 1))
```

Comment:
Does this handle dynamic shapes though? You might need to check the size using V.graph.sizevars (search for that string elsewhere in the codebase)